rust/rbac-registration/src/cardano/cip509/utils/extract_key.rs

```diff
@@ -22,24 +78,19 @@ pub struct SignatureAlgoError {
 ///
 /// Returns an error if the signature algorithm is not supported and
 /// the public key cannot be extracted.
-pub fn x509_key(cert: &X509Certificate) -> anyhow::Result<VerifyingKey> {
-    let oid: Oid = cert
-        .tbs_certificate
-        .subject_public_key_info
-        .algorithm
-        .oid
-        .to_string()
-        .parse()
-        // `Context` cannot be used here because `OidParseError` doesn't implement `std::Error`.
-        .map_err(|e| anyhow!("Invalid signature algorithm OID: {e:?}"))?;
-    check_signature_algorithm(&oid)?;
-    let extended_public_key = cert
+///
+/// Returns an error if public key has unexpected value.
+pub fn x509_key(cert: &X509Certificate) -> Result<VerifyingKey, Error> {
+    let oid = &cert.tbs_certificate.subject_public_key_info.algorithm.oid;
+    check_signature_algorithm(&spki_oid_as_asn1_rs_oid(oid))?;
+    let public_key = &cert
         .tbs_certificate
         .subject_public_key_info
-        .subject_public_key
-        .as_bytes()
-        .context("Invalid subject_public_key value (has unused bits)")?;
-    verifying_key(extended_public_key).context("Unable to get verifying key from X509 certificate")
+        .subject_public_key;
+    let public_key_bytes = public_key.as_bytes().ok_or(Error::InvalidPublicKeyLength {
```

Comment:
Nitpick.
Seems like not a totally correct reporting error variant/message. Should be something about not correctly aligned/has unused bits.

From the `as_bytes` docs

```
        /// Borrow the inner byte slice.
        ///
        /// Returns `None` if the number of unused bits is *not* equal to zero,
        /// i.e. if the `BIT STRING` is not octet aligned.
        ///
        /// Use [`BitString::raw_bytes`] to obtain access to the raw value
        /// regardless of the presence of unused bits.
        pub fn as_bytes(&self) -> Option<&[u8]> {
```